File: 2/calc-1.clj
(defn parse [xs]
  (map #(Integer/parseInt %) xs))

(defn calc-score [scores opponent us]
  (let [points-from-play (scores us)
        points-from-result (cond
                             (= (scores opponent) (scores us)) (scores :draw)
                             (or (and (= us "X") (= opponent "C"))
                                 (and (= us "Y") (= opponent "A"))
                                 (and (= us "Z") (= opponent "B"))) (scores :win)
                             :else (scores :lose))]
    (+ points-from-play points-from-result)))

(let [input (slurp "input-1.txt")
      scores {"A" 1 "B" 2 "C" 3 "X" 1 "Y" 2 "Z" 3 :win 6 :draw 3 :lose 0}]
  (as-> input v
    (str/split v #"\n") ;; split into lines
    (map #(str/split % #" ") v)
    (map #(calc-score scores (first %) (second %)) v)
    (reduce + v)
    (println v)))

File: 2/calc-1.js
const fs = require("fs");
const { reduce } = require("lodash");

const input = fs.readFileSync("./input-1.txt").toString();

const scores = {
  A: 1, // rock
  B: 2, // paper
  C: 3, // scissors
  X: 1, // rock
  Y: 2, // paper
  Z: 3, // scissors
  lose: 0,
  draw: 3,
  win: 6,
};

const rounds = input.split("\n").slice(0, -1);

const points = rounds.map((round) => {
  let points = 0;
  const [team1, team2] = round.split(" ");
  points += scores[team2];
  if (scores[team1] === scores[team2]) {
    points += scores.draw;
  }
  if (
    (team2 === "X" && team1 === "C") ||
    (team2 === "Y" && team1 === "A") ||
    (team2 === "Z" && team1 === "B")
  ) {
    points += scores.win;
  }
  return points;
});

const totalPoints = points.reduce((acc, curr) => acc + curr, 0);

console.log(totalPoints);
